### PR TITLE
repoman: add a check for unsorted KEYWORDS

### PR DESCRIPTION
**Pull request by sbraz**
I thought @SoapGentoo would enjoy that kind of check :)

### CODE REVIEW

**Comment by dol-sen** [comment]
hmm, this is good, but I am wondering if repoman shouldn't just auto-sort the entries like it does the copyright update.   Make it a warn that it found unsorted keywords (and fixed).

File: repoman/pym/repoman/modules/scan/keywords/keywords.py [hunk 1]
```diff
@@ -133,6 +135,16 @@ def _checkForMaskLikeKeywords(self, xpkg, y_ebuild, keywords):
 				self.qatracker.add_error("KEYWORDS.stupid",
 					"%s/%s.ebuild" % (xpkg, y_ebuild))
 
+	def _checkForUnsortedKeywords(self, ebuild, xpkg, y_ebuild):
+		'''Ebuilds that contain KEYWORDS
+		which are not sorted alphabetically.'''
+		sorted_keywords = sorted(ebuild.keywords,
+			key=lambda x: x.lstrip("-~"))
```
Comment (Arfrever) [inline repoman/pym/repoman/modules/scan/keywords/keywords.py]: This is broken implementation of sorting.

`ekeyword` tool uses https://gitweb.gentoo.org/proj/gentoolkit.git/tree/pym/gentoolkit/ekeyword/ekeyword.py?id=bb585671418fa51682a7e1dbfef5914e26261adf#n82 which places `*-*` keywords after other keywords, so e.g. `~amd64 ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux` is properly sorted value.

Comment (SoapGentoo) [reply repoman/pym/repoman/modules/scan/keywords/keywords.py]: Then send in a fix instead of just peanut gallery'ing

**Comment by sbraz** [comment]
@Arfrever fixed. This will behave like ekeyword.

**Comment by sbraz** [comment]
@Arfrever re-pushed and this time it matches the behaviour of ekeyword, can you confirm?

**Comment by Arfrever** [comment]
It seems to work.

**Comment by zmedico** [comment]
Merged in f4faf6d7950150a3068f92a3f3615b884db897d6, thanks!

**Pull request closed without merging**